akka-persistence-typed/src/main/scala/akka/persistence/typed/Retention.scala

```diff
@@ -34,22 +35,26 @@ final case class RetentionCriteria(snapshotEveryNEvents: Long, keepNSnapshots: L
     // Delete old events, retain the latest
     math.max(0, lastSequenceNr - (keepNSnapshots * snapshotEveryNEvents))
   }
+
+  /** Java API. */
+  def withKeepNSnapshots(keepN: Long): RetentionCriteria =
+    copy(keepNSnapshots = keepN)
+
+  /** Java API. */
+  def withDeleteEventsOnSnapshot(): RetentionCriteria =
+    copy(deleteEventsOnSnapshot = true)
 }
 
 object RetentionCriteria {
 
-  def apply(): RetentionCriteria =
+  def default: RetentionCriteria =
```

Comment:
since it is immutable it can be a val to avoid allocating a new one for every use.

Reply:
Good catch, it was in an earlier iteration.